lib/child_process.js

```diff
@@ -679,6 +702,33 @@ function sanitizeKillSignal(killSignal) {
   }
 }
 
+
+function quotePosixShArg(arg) {
+  return `'${arg.replace(/'/g, "'\\''")}'`;
```

Comment:
```suggestion
  return `'${StringPrototypeReplace(arg, /'/g, "'\\''")}'`;
```
For info, we can't use `replaceAll` from primordials yet unfortunately.